apps/web/src/components/invoice-generator/index.tsx

```diff
@@ -0,0 +1,478 @@
+"use client";
+
+/* eslint-disable no-nested-ternary -- this is pretty cool and readable here */
+
+import { Button, Checkbox, Input, Textarea } from "@tietokilta/ui";
+// eslint-disable-next-line import/named -- Next.js magic enables this
+import { useFormState, useFormStatus } from "react-dom";
+import {
+  type FormEventHandler,
+  type HTMLInputAutoCompleteAttribute,
+  type HTMLInputTypeAttribute,
+  type ReactNode,
+  useEffect,
+  useState,
+} from "react";
+import {
+  I18nProviderClient,
+  useCurrentLocale,
+  useScopedI18n,
+} from "../../locales/client";
+import { SaveAction } from "../../lib/api/external/laskugeneraattori/actions";
+import { type InvoiceGeneratorFormState } from "../../lib/api/external/laskugeneraattori/index";
+
+interface GenericFieldProps {
```

Comment:
It's not a big deal here, but it seems we are mostly extending the native input props, so we could probably directly extend it with `extends InputHTMLAttributes<HTMLInputElement>`